run.py

```diff
@@ -290,6 +290,51 @@ def delete_obsolete_records(older_than, batch_size=1000):
 
     logger.info("Deleted {} obsolete records".format(deleted))
 
+def process_all_scixid(batch_size, flag):
```

Comment:
This looks good. I think we can strip out the queue length monitoring, though. We only really do that for reindexing so we can tell when it is time to swap the solr cores. This should be fine to let run asynchronously.